bundle-types/action.yml

```diff
@@ -0,0 +1,116 @@
+name: "Grafana Bundle Types"
+description: "Bundles a Grafana plugins types and pushes them to npm."
+
+outputs:
+  typesFilePath:
+    description: "The path to the built types file."
+    value: ${{ steps.set-outputs.outputs.typesFilePath }}
+
+inputs:
+  entryPoint:
+    description: "Location of types file to bundle."
+    required: false
+    default: "./src/types/index.ts"
+  tsConfig:
+    description: "A path to the tsconfig file to use when bundling types."
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Install dependencies
+      run: ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Get plugin id and version
+      id: metadata
+      run: |
+        PLUGIN_ID=$(cat src/plugin.json | jq -r .id)
+        echo "plugin-id=${PLUGIN_ID}" >> $GITHUB_OUTPUT
+        PLUGIN_VERSION=$(cat package.json | jq -r .version)
+        echo "plugin-version=${PLUGIN_VERSION}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - id: get-secrets
+      uses: grafana/shared-workflows/actions/get-vault-secrets@main
+      with:
+        common_secrets: |
+          GITHUB_APP_ID=plugins-platform-bot-app:app-id
+          GITHUB_APP_PRIVATE_KEY=plugins-platform-bot-app:private-key
+
+    - name: Generate token
+      id: generate-token
+      uses: tibdex/github-app-token@v2
+      with:
+        app_id: ${{ env.GITHUB_APP_ID }}
+        private_key: ${{ env.GITHUB_APP_PRIVATE_KEY }}
+
+    - name: Clone NPM types package repo
+      run: |
+        cd ..
+        gh auth status
```

Comment:
is this command required?

Reply:
It's not required but I felt it could be useful for debug purposes? Can always remove it if we feel it's worthless.

```
github.com
  ✓ Logged in to github.com account grafana-plugins-platform-bot[bot] (GITHUB_TOKEN)
  - Active account: true
  - Git operations protocol: https
  - Token: ghs_************************************
 ```